soc/xtensa/dc233c/include/xtensa-dc233c.ld

```diff
@@ -0,0 +1,501 @@
+  sram0_14_seg : org = 0x00002300, len = 0x38
+  sram0_15_seg : org = 0x00002338, len = 0x8
+  sram0_16_seg : org = 0x00002340, len = 0x38
+  sram0_17_seg : org = 0x00002378, len = 0x48
+  sram0_18_seg : org = 0x000023C0, len = 0x40
+#ifdef CONFIG_XTENSA_MMU
+  vec_helpers  : org = 0x00002400, len = (PHYS_RAM_ADDR - 0x00002400)
+#endif
+  RAM          : org = PHYS_RAM_ADDR, len = PHYS_RAM_SIZE
+
+  /* Although ROM is of size 0x02000000, we limit it to 8KB so
+   * fewer L2 page table entries are needed.
+   */
+  rom0_seg    : org = PHYS_ROM0_ADDR, len = PHYS_ROM0_SIZE
+
+#ifdef CONFIG_GEN_ISR_TABLES
+  /* The space before exception vectors is not being used.
+   * So we stuff the temporary IDT_LIST there to avoid
+   * some linker issues which would balloon the size of
+   * the intermediate files (like zephyr_pre0.elf, to
+   * couple hundred MBs or even GBs).
+   */
+  IDT_LIST     : org = 0x00000000, len = 0x2000
+#endif
+}
+
+PHDRS
+{
+  sram0_0_phdr PT_LOAD;
+  sram0_1_phdr PT_LOAD;
+  sram0_2_phdr PT_LOAD;
+  sram0_3_phdr PT_LOAD;
+  sram0_4_phdr PT_LOAD;
+  sram0_5_phdr PT_LOAD;
+  sram0_6_phdr PT_LOAD;
+  sram0_7_phdr PT_LOAD;
+  sram0_8_phdr PT_LOAD;
+  sram0_9_phdr PT_LOAD;
+  sram0_10_phdr PT_LOAD;
+  sram0_11_phdr PT_LOAD;
+  sram0_12_phdr PT_LOAD;
+  sram0_13_phdr PT_LOAD;
+  sram0_14_phdr PT_LOAD;
+  sram0_15_phdr PT_LOAD;
+  sram0_16_phdr PT_LOAD;
+  sram0_17_phdr PT_LOAD;
+  sram0_18_phdr PT_LOAD;
+
+#ifdef CONFIG_XTENSA_MMU
+  vec_helpers_phdr PT_LOAD;
+#endif
+
+  rom0_phdr PT_LOAD;
+  sram0_phdr PT_LOAD;
+  sram0_bss_phdr PT_LOAD;
+}
+
+
+/*  Default entry point:  */
+ENTRY(CONFIG_KERNEL_ENTRY)
+
+_rom_store_table = 0;
+PROVIDE(_memmap_vecbase_reset = 0x00002000);
+PROVIDE(_memmap_reset_vector = 0xFE000000);
+/* Various memory-map dependent cache attribute settings:
+ *
+ * Note that there is no cacheattr register which means thah
+ * cacheattr is emulated through TLB way 6 (8x 512MB regions).
+ * So the attributes here are the MMU memory attributes:
+ *   0x3 - rwx, bypass cache
+ *   0x7 - rwx, cache write back
+ *   0xB - wrx, cache write through
+ * Refer to the ISA manual for other attributes.
+ */
+_memmap_cacheattr_wb_base = 0x70000007;
+_memmap_cacheattr_wt_base = 0xB000000B;
+_memmap_cacheattr_bp_base = 0x30000003;
+_memmap_cacheattr_unused_mask = 0x0FFFFFF0;
+_memmap_cacheattr_wb_strict = 0x7FFFFFF7;
+_memmap_cacheattr_wt_strict = 0xBFFFFFFB;
+_memmap_cacheattr_bp_strict = 0x3FFFFFF3;
+_memmap_cacheattr_wb_allvalid = 0x73333337;
+_memmap_cacheattr_wt_allvalid = 0xB333333B;
+_memmap_cacheattr_bp_allvalid = 0x33333333;
+PROVIDE(_memmap_cacheattr_reset = _memmap_cacheattr_wb_allvalid);
+
+SECTIONS
+{
+
+#include <zephyr/linker/rel-sections.ld>
+
+#ifdef CONFIG_GEN_ISR_TABLES
+#include <zephyr/linker/intlist.ld>
+#endif
+
+  .WindowVectors.text : ALIGN(4)
+  {
+    _WindowVectors_text_start = ABSOLUTE(.);
+    KEEP (*(.WindowVectors.text))
+    _WindowVectors_text_end = ABSOLUTE(.);
+  } >sram0_0_seg :sram0_0_phdr
```

Comment:
Likewise there's a patch in the mt8195 series which will auto-generate all this vector nonsense for you from the offsets in the core-isa.h file, putting it in a file you can just include that emits to just one memory region.  Saves a ton of boilerplate.